GitUI/UserControls/FileStatusList.cs

```diff
@@ -92,31 +94,45 @@ ImageList CreateImageList()
-                        ScaleHeight(Images.FileStatusModified), // 2
-                        ScaleHeight(Images.FileStatusRenamed.AdaptLightness()), // 3
-                        ScaleHeight(Images.FileStatusCopied), // 4
-                        ScaleHeight(Images.SubmoduleDirty), // 5
-                        ScaleHeight(Images.SubmoduleRevisionUp), // 6
-                        ScaleHeight(Images.SubmoduleRevisionUpDirty), // 7
-                        ScaleHeight(Images.SubmoduleRevisionDown), // 8
-                        ScaleHeight(Images.SubmoduleRevisionDownDirty), // 9
-                        ScaleHeight(Images.SubmoduleRevisionSemiUp), // 10
-                        ScaleHeight(Images.SubmoduleRevisionSemiUpDirty), // 11
-                        ScaleHeight(Images.SubmoduleRevisionSemiDown), // 12
-                        ScaleHeight(Images.SubmoduleRevisionSemiDownDirty), // 13
-                        ScaleHeight(Images.FileStatusUnknown) // 14
-                    }
                 };
 
-                Bitmap ScaleHeight(Bitmap input)
+                _stateImageIndexDict = new Dictionary<string, int>();
+                var images = new (string imageKey, Bitmap icon)[]
+                {
+                    (nameof(Images.FileStatusUnknown), Images.FileStatusUnknown),
+                    (nameof(Images.FileStatusModified), Images.FileStatusModified),
+                    (nameof(Images.FileStatusAdded), Images.FileStatusAdded),
+                    (nameof(Images.FileStatusRemoved), Images.FileStatusRemoved),
+                    (nameof(Images.Conflict), Images.Conflict),
+                    (nameof(Images.FileStatusRenamed), Images.FileStatusRenamed.AdaptLightness()),
+                    (nameof(Images.FileStatusCopied), Images.FileStatusCopied),
+                    (nameof(Images.SubmodulesManage), Images.SubmodulesManage),
+                    (nameof(Images.FolderSubmodule), Images.FolderSubmodule),
+                    (nameof(Images.SubmoduleDirty), Images.SubmoduleDirty),
+                    (nameof(Images.SubmoduleRevisionUp), Images.SubmoduleRevisionUp),
+                    (nameof(Images.SubmoduleRevisionUpDirty), Images.SubmoduleRevisionUpDirty),
+                    (nameof(Images.SubmoduleRevisionDown), Images.SubmoduleRevisionDown),
+                    (nameof(Images.SubmoduleRevisionDownDirty), Images.SubmoduleRevisionDownDirty),
+                    (nameof(Images.SubmoduleRevisionSemiUp), Images.SubmoduleRevisionSemiUp),
+                    (nameof(Images.SubmoduleRevisionSemiUpDirty), Images.SubmoduleRevisionSemiUpDirty),
+                    (nameof(Images.SubmoduleRevisionSemiDown), Images.SubmoduleRevisionSemiDown),
+                    (nameof(Images.SubmoduleRevisionSemiDownDirty), Images.SubmoduleRevisionSemiDownDirty)
+                };
+
+                var i = 0;
+                foreach ((string imageKey, Bitmap icon) in images)
```

Comment:
Suggest converting to a for loop instead of foreach. This way you're guaranteed to have the right index.

Reply:
ok